contracts/modules/cover/Gateway.sol

```diff
@@ -173,10 +182,19 @@ contract Gateway is MasterAware {
   {
     (, uint coverId) = claimsData.getClaimCoverId(claimId);
     (, uint internalClaimStatus) = claimsData.getClaimStatusNumber(claimId);
+    (,address productId) = quotationData.getscAddressOfCover(coverId);
+    address coveredTokenAddress = incidents.coveredToken(productId);
```

Comment:
The `coveredTokenAddress` and `productId` reads need to happen conditionally. We are not always using their value.